postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.postgresql.util.PGInterval;
+
+import java.lang.reflect.Type;
+import java.sql.Types;
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Postgres version of argument factory for {@link Duration}.
+ *
+ * <p>
+ * For simplicity, this implementation makes the duration positive before proceeding. However, this can cause an
+ * {@link ArithmeticException} to be thrown. E.g., this can occur if your duration is -2^63 seconds.
+ *
+ * <p>
+ * Not all {@link Duration}s can be represented as intervals in Postgres.
+ * One incompatibility results from {@link Duration}s that are too large. This is due to (1) {@link Duration}s using
+ * a {@code long} internally, and {@link PGInterval}s using {@code int}s; and (2) the conversion of days to months or
+ * years being ambiguous.
+ * Another results from {@link Duration}s being too precise; they have nanosecond precision, whereas Postgres has only
+ * microsecond.
+ * An {@link IllegalArgumentException} will be thrown in these cases.
+ * The handling of the second is subject to revision in the future; for example, it would be reasonable to have a
+ * configurable truncation option.
+ */
+public class DurationArgumentFactory implements ArgumentFactory {
+    @Override
+    public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
+        if (!(value instanceof Duration)) {
+            return Optional.empty();
+        }
```

Comment:
Oh whoops.  I missed your suggestion of the right way to return `null`.  Fixed in 5ace9385f059734ea6aae3bc99c3b36ee31ab7cc.